service/src/main/java/uk/gov/hmcts/reform/fpl/service/validators/EventsChecker.java

```diff
@@ -0,0 +1,127 @@
+import static uk.gov.hmcts.reform.fpl.enums.Event.OTHERS;
+import static uk.gov.hmcts.reform.fpl.enums.Event.OTHER_PROCEEDINGS;
+import static uk.gov.hmcts.reform.fpl.enums.Event.RESPONDENTS;
+import static uk.gov.hmcts.reform.fpl.enums.Event.RISK_AND_HARM;
+import static uk.gov.hmcts.reform.fpl.enums.Event.SUBMIT_APPLICATION;
+
+@Service
+public class EventsChecker {
+
+    @Autowired
+    private CaseNameChecker caseNameChecker;
+
+    @Autowired
+    private ChildrenChecker childrenChecker;
+
+    @Autowired
+    private RespondentsChecker respondentsChecker;
+
+    @Autowired
+    private HearingUrgencyChecker hearingUrgencyChecker;
+
+    @Autowired
+    private OrdersSoughtChecker ordersSoughtChecker;
+
+    @Autowired
+    private GroundsChecker groundsChecker;
+
+    @Autowired
+    private OrganisationDetailsChecker organisationDetailsChecker;
+
+    @Autowired
+    private AllocationProposalChecker allocationProposalChecker;
+
+    @Autowired
+    private DocumentsChecker documentsChecker;
+
+    @Autowired
+    private CaseSubmissionChecker caseSubmissionChecker;
+
+    @Autowired
+    private RiskAndHarmChecker riskAndHarmChecker;
+
+    @Autowired
+    private ProceedingsChecker proceedingsChecker;
+
+    @Autowired
+    private InternationalElementChecker internationalElementChecker;
+
+    @Autowired
+    private OthersChecker othersChecker;
+
+    @Autowired
+    private CourtServiceChecker courtServiceChecker;
+
+    @Autowired
+    private FactorsAffectingParentingChecker factorsAffectingParentingChecker;
+
+    private final EnumMap<Event, EventChecker> eventCheckers = new EnumMap<>(Event.class);
+
+    @PostConstruct
+    public void init() {
+        eventCheckers.put(CASE_NAME, caseNameChecker);
+        eventCheckers.put(CHILDREN, childrenChecker);
+        eventCheckers.put(RESPONDENTS, respondentsChecker);
+        eventCheckers.put(HEARING_URGENCY, hearingUrgencyChecker);
+        eventCheckers.put(ORDERS_SOUGHT, ordersSoughtChecker);
+        eventCheckers.put(GROUNDS, groundsChecker);
+        eventCheckers.put(ORGANISATION_DETAILS, organisationDetailsChecker);
+        eventCheckers.put(ALLOCATION_PROPOSAL, allocationProposalChecker);
+        eventCheckers.put(DOCUMENTS, documentsChecker);
+        eventCheckers.put(RISK_AND_HARM, riskAndHarmChecker);
+        eventCheckers.put(FACTORS_AFFECTING_PARENTING, factorsAffectingParentingChecker);
+        eventCheckers.put(OTHER_PROCEEDINGS, proceedingsChecker);
+        eventCheckers.put(INTERNATIONAL_ELEMENT, internationalElementChecker);
+        eventCheckers.put(OTHERS, othersChecker);
+        eventCheckers.put(COURT_SERVICES, courtServiceChecker);
+        eventCheckers.put(SUBMIT_APPLICATION, caseSubmissionChecker);
+    }
+
+    public List<String> validate(Event event, CaseData caseData) {
+        return ofNullable(eventCheckers.get(event))
+                .map(validator -> validator.validate(caseData))
+                .orElse(emptyList());
+    }
```

Comment:
So we have this `ValidateGroupService` with a `validateGroup`, maybe I missed this in the original PR but why did we not re-use?

Reply:
ValidateGroupService validates entire case and returns list of all errors for entire case. Here we need errors for part of case data which are modified by particular event (e.g if event is AmendChildren, we need to get errors only for Children and not for entire case)

Reply:
Could we have solved this by creating more validation groups perhaps? That way we could of used `ValidateGroupService(caseData, childrenGroup.class)`